### PR TITLE
Entity API tests for the papiea client in typescript sdk

### PR DESCRIPTION
**Pull request by nitesh-idnani1**

### CODE REVIEW

**Review by Emulebest — COMMENTED** [review]
To be honest, I'd rather we make a simple scenario for this, rather than copying the functionality of engine tests. Otherwise, we would need to implement the same case for other client as well. 

Also is yarn-error.log file needed in the PR?

**Comment by nitesh-idnani1** [comment]
> To be honest, I'd rather we make a simple scenario for this, rather than copying the functionality of engine tests. Otherwise, we would need to implement the same case for other client as well.
> 
> Also is yarn-error.log file needed in the PR?

Yes, I'm working on the next steps for this which will include a e2e test scenario similar to what we have in python e2e tests. We can discuss more on this is the meeting.
Also, removed the yarn-error log file, it was added by mistake.

File: papiea-sdk/typescript/__tests__/client_tests/e2e_tests/test_setup.ts [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+const metadata_extension = load(readFileSync(resolve(__dirname, "./security/metadata_extension.yml"), "utf-8"));
+
+const bucket_yaml = load(readFileSync(resolve(__dirname, "./kinds/bucket_kind.yml"), "utf-8"));
+bucket_yaml["bucket"]["properties"]["objects"]["items"]["properties"]["reference"] =
+    utils.get_kind_ref_type(config.object, "Reference of the objects within the bucket")
+
+const object_yaml = load(readFileSync(resolve(__dirname, "./kinds/object_kind.yml"), "utf-8"));
+object_yaml["object"]["properties"]["references"]["items"]["properties"]["bucket_reference"] =
+    utils.get_kind_ref_type(config.bucket, "Reference of the bucket in which the object exists")
```
Comment (shlomi-nx) [inline papiea-sdk/typescript/__tests__/client_tests/e2e_tests/test_setup.ts:23]: Suggestion: Although convenient to use here, perhaps one of these should remain in the yaml to serve as a dual example - how to do that directly in YAML, and the benefits of doing through code

**Review by shlomi-nx — COMMENTED** [review]
Looks very good to me. A couple of minor comments.

Perhaps we should extract this e2e provider as an example provider, and have the e2e use it. This way we can add this as a canonical example for papiea, and have the Readme point to it.

File: papiea-client/src/entity_client.ts [hunk 2]
```diff
@@ -142,10 +142,11 @@ export async function filter_entity_iter(provider: string, kind: string, version
     return iter_func
 }
 
-export async function filter_entity(provider: string, kind: string, version: string, filter: any, papiea_url: string, s2skey: string, tracer: Tracer): Promise<FilterResults> {
+export async function filter_entity(provider: string, kind: string, version: string, filter: any, query: string, papiea_url: string, s2skey: string, tracer: Tracer): Promise<FilterResults> {
     const headers = getHeaders(s2skey)
     const span = spanOperation("filter_entity_client", {headers, tracer})
-    const res = await make_request(axios.post, `${ papiea_url }/services/${ provider }/${ version }/${ kind }/filter`, filter, {headers});
+    const query_val: string = (query === '' ? query : '?' + query);
+    const res = await make_request(axios.post, `${ papiea_url }/services/${ provider }/${ version }/${ kind }/filter${ query_val }`, filter, {headers});
```
[the remark this answers is not in the document]
Comment (nitesh-idnani1) [reply papiea-client/src/entity_client.ts:149]: The query was already a part of the papiea engine implementation, but there was no support to set it through the client.
These are the permissible query params supported now
https://github.com/nutanix/papiea/blob/c8f8e3b6fc8a03301aa13ba073373a3ca6d522a1/papiea-engine/src/entity/entity_routes.ts#L135

**Review by Emulebest — CHANGES_REQUESTED** [review]
One small change. I think it is better to have directory structure of:
`__tests__`
      `provider_sdk_tests`
      `intentful_workflow_tests`
      `e2e_tests`

Other than that feel free to merge

**Pull request merged**